clients/dos/src/main/kotlin/DosClient.kt

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2024 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.clients.dos
+
+import java.io.File
+
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.RequestBody.Companion.asRequestBody
+
+import org.apache.logging.log4j.kotlin.logger
+
+import org.ossreviewtoolkit.utils.common.formatSizeInMib
+
+/**
+ * A client implementation on top of the DOS service.
+ */
+class DosClient(private val service: DosService) {
+    /**
+     * Get the package configuration for the given [purl]. Return a [PackageConfigurationResponseBody] or null on error.
+     */
+    suspend fun getPackageConfiguration(purl: String): PackageConfigurationResponseBody? {
+        if (purl.isEmpty()) {
+            logger.error { "The Package URL is required to get the package configuration." }
+            return null
+        }
+
+        val requestBody = PackageConfigurationRequestBody(purl)
+        val response = service.getPackageConfiguration(requestBody)
+        val responseBody = response.body()
+
+        return if (response.isSuccessful && responseBody != null) {
+            responseBody
+        } else {
+            logger.error { "Error getting the package configuration for PURL $purl: ${response.errorBody()}" }
+
+            null
+        }
+    }
+
+    /**
+     * Return a pre-signed URL as provided by the DOS API to upload a package to S3 object storage for scanning, or null
+     * if the request did not succeed.
+     */
+    suspend fun getUploadUrl(key: String): String? {
+        if (key.isEmpty()) {
+            logger.error { "Need the name of the zipped packet to upload" }

```

Comment:
Is `packet` correct? 

Missing trailing dot.

Reply:
I'll fix these in a follow-up, thanks.

Reply:
See https://github.com/oss-review-toolkit/ort/pull/8789.